marginalia.el

```diff
@@ -1113,7 +1113,12 @@ These annotations are skipped for remote paths."
 (defun marginalia-classify-by-command-name ()
   "Lookup category for current command."
   (and marginalia--command
-       (alist-get marginalia--command marginalia-command-categories)))
+       (or (alist-get marginalia--command marginalia-command-categories)
+           ;; The command could also be an alias to a command in
+           ;; `marginalia-command-categories', e.g., `recentf' is an alias for
+           ;; `recentf-open'.
+           (when-let ((chain (compat-call function-alias-p marginalia--command)))
```

Comment:
Thanks. The compat-call is unnecessary. It looks good otherwise.

Reply:
Why is the `compat-call` unnecessary?  `function-alias-p` is new in emacs 29.1 so I can't just call it, I think.  But please tell me if my understanding of compat is wrong.

Yes, `chain` could be inlined but then `marginalia-command-categories` would always be traversed looking for an entry with `nil` key.  No big deal as long as the list doesn't become large.  But still it seemed like a better idea to do nothing if there's no chain given that the function is easily called 80 times during a not so complex find-file operation.

Reply:
> But please tell me if my understanding of compat is wrong.

Yes, your understanding is wrong. Please read the Compat manual and tell me how you came to the wrong conclusion. Then we can hopefully improve the manual.

> But still it seemed like a better idea to do nothing if there's no chain given that the function is easily called 80 times during a not so complex find-file operation.

These lists are usually short, so it won't matter. But your argument is good in a semantic sense - it is simply wrong to look for the nil key.